keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1066 @@
+immediately upon first becoming aware of a `Type: LoadBalancer`
+Service. At all times, the condition should be `True` if the load
+balancer is handling connections, and `False` if it is not. (There are
+no defined semantics for an `Unknown` value of this condition.)
+
+(This means that at construction time, the `LoadBalancerServing`
+condition is redundant with the `.Status.LoadBalancer` field; it
+should switch from `False` to `True` at exactly the same time as the
+`.Status.LoadBalancer` field is filled in. However, if the Service is
+updated later, the cloud provider may temporarily switch it to
+`LoadBalancerServing=False` while reprovisioning the Service, and it
+would not generally clear `.Status.LoadBalancer` during this time.)
+
+Any time the load balancer behavior changes in an externally-visible
+way, the cloud provider must update at least the `LastTransitionTime`
+of the `LoadBalancerServing` condition. (For example, if the user
+changes the service's `LoadBalancerSourceRanges`, and the cloud
+provider updates the load balancer for this, it must update the
+`LoadBalancerServing` condition's `LastTransitionTime`, even if the
+value of `LoadBalancerServing` did not toggle from `True` to `False`
+and back to `True`.)
+
+The well-known `Reason` values for the `LoadBalancerServing` condition
+are:
+
+  - `Status=True, Reason=Serving`
+
+  - `Status=False, Reason=Provisioning`: The load balancer is not
+    currently serving because it is being provisioned /
+    re-provisioned.
+
+  - `Status=False, Reason=Unsupported`: No load balancer could be
+    provisioned, because the cloud provider doesn't support some
+    feature of this Service (explained further in `Message`).
+
+  - `Status=False, Reason=Infrastructure`: No load balancer could be
+    provisioned, because of a problem with the underlying cloud
+    infrastructure (explained further in `Message`).
+
+#### The `LoadBalancerProvisioning` Condition
+
+The `LoadBalancerProvisioning` condition, when `True`, indicates that
+the cloud provider is currently performing a time-consuming
+resource-provisioning step for the load balancer. When
+`LoadBalancerProvisioning` is `True`, the load balancer is either not
+handling connections, or else is still handling connections according
+to the semantics of a previous state of the Service (with the value of
+the `LoadBalancerServing` condition distinguishing these two cases).
+
+The cloud provider should set a `LoadBalancerProvisioning` condition
+immediately upon first becoming aware of a `Type: LoadBalancer`
+Service. At all times, the condition should be `True` if the cloud
+provider is waiting for the completion of an operation to bring the
+load balancer's behavior in line with the current state of the Service
+object, and `False` if it not. (There are no defined semantics for an
+`Unknown` value of this condition.)
+
+Any time any load-balancer-related field (or provider-specific
```

Comment:
other fields are not really timestamped, so unless you know what time you made a change, this semantic is meaningless.  If you neeeeed to know this, we should set something atomically in the REST handler and force providers to clear it (same as we are doing for in-place pod updates)